src/server/cluster/incoming_slot_migration.cc

```diff
@@ -125,11 +125,10 @@ class ClusterShardMigration {
   }
 
  private:
-  void ExecuteTxWithNoShardSync(TransactionData&& tx_data, Context* cntx) {
+  void ExecuteTx(TransactionData&& tx_data, Context* cntx) {
     if (cntx->IsCancelled()) {
       return;
     }
-    CHECK(tx_data.shard_cnt <= 1);  // we don't support sync for multishard execution
```

Comment:
It took me a while to understand how we dont fail here in our tests and why we have this check.
I now understand that in cluster shard by tag and therefore a transaction runs on once shard
https://github.com/dragonflydb/dragonfly/blob/14fda354ccdffcf33df7c2af04937051382ead74/src/server/cluster/cluster_defs.cc#L116
We can add a unit test to make sure we dont break this